analyses.go

```diff
@@ -152,6 +152,95 @@ func JobsToKill(ctx context.Context, dedb *sql.DB) ([]Job, error) {
 	return jobs, nil
 }
 
+const periodicWarningsQuery = `
+SELECT jobs.id,
+       jobs.app_id,
+       jobs.user_id,
+       jobs.status,
+       jobs.job_description,
+       jobs.job_name,
+       jobs.result_folder_path,
+       jobs.planned_end_date,
+       jobs.start_date,
+       job_types.system_id,
+       users.username
+  FROM jobs
+  JOIN job_types on jobs.job_type_id = job_types.id
+  JOIN users on jobs.user_id = users.id
+  LEFT join notif_statuses ON jobs.id = notif_statuses.analysis_id
+ WHERE jobs.status = $1
+   AND (notif_statuses.last_periodic_warning is null
+    OR notif_statuses.last_periodic_warning < $2 - (coalesce(notif_statuses.periodic_warning_period, 14400)::text || ' seconds'::text)::interval)
```

Comment:
On those two thoughts:
a.) I didn't use `now` or `CURRENT_TIMESTAMP` (equivalent) mostly because I wanted to be able to use the same timestamp for everything, but looking at it, this function is using its own call to `time.Now` and then the loop in main.go is the only one actually reusing it. So I think I agree and we can switch this to that, just to remove another place something can go wrong.
b.) I like this, though I think it basically just means we use string concatenation on the insert end instead. Still, that should mean the query that's called much more often (this one) is better.

Reply:
I put up https://github.com/cyverse-de/de-database/pull/26 to change that type. I'll look at the updates that need to happen here and try to get those up in this PR concurrently.

Reply:
Sounds good. Yeah, unfortunately, I don't see a good way to eliminate the string concatenation from both SQL statements either.